3scale/porta
# Implements Vertical Nav in Patternfly 4

**What this PR does / why we need it**:

Uses Patternfly 4 React to redesign the vertical navigation.

Changes all slim "sections" logic into a ruby helper.
Implements a Vertical Navigation component in React using Patternfly 4 styles.
Instead of using PF4-React it uses a "proxy" class that uses its same structure so that it could be easily swapped.

Look & Feel (ContextSelector icon takes a bit to load)
![vertnav01](https://user-images.githubusercontent.com/11672286/59265598-f6a2b080-8c45-11e9-9bd4-6f476cfb4d7a.gif)


**Which issue(s) this PR fixes** 

[THREESCALE-2603: Add PF4 vertical navbar](https://issues.jboss.org/browse/THREESCALE-2603)

**Verification steps** 

Navigate to all possible sections and verify everything works as expected and items highlights

**NOTES**
Patternfly fonts are not being used so probably there are more things that could be removed from assets/patternfly. This is to be investigated.
